### ISSUE TITLE
cannot enable cl_khr_fp_16 extension on this platform

### ISSUE THREAD

**Issue by maded2** [exo-explore/exo#583]
get this error when trying to load a model

my GPU: Linux Box (NVIDIA GEFORC/ GTX 1080 TI) 10GB

**Comment by david-thrower** [comment]
## TLDR: This may be a structural issue: 

This is a first pass answer. There's no promise this is the reason in your case, but I found this:

The cl_khr_fp16 extension, which enables half-precision (FP16) support in OpenCL, is not fully supported on many NVIDIA GPUs, **despite** the hardware capabilities. This is apparently a long-standing issue for OpenCL when trying to utilize FP16 operations on NVIDIA.

## A few quick points:

1. NVIDIA does not officially report cl_khr_fp16 as a supported extension in their OpenCL implementations[1].

2. Even when the extension appears to be enabled, certain FP16 operations may still fail or produce unexpected results[1].

If I had to guess, it is probably Tiny Grad or llama.cpp that as the intermediate component to blame, but I don't know enough details about what you did, what model you used, what options, env vars, flags, etc ... you have set, so this is just a guess.

[1] https://forums.developer.nvidia.com/t/cl-khr-fp16-opencl-support/191106

**Comment by octaveoclx** [comment]
https://developer.nvidia.com/bugs/4152297 May need more people's voice. Currently, it can pass clblast which only used + - * /. 
Math functions are missing.
